File: problem 12/solution.c
#include <stdio.h>
#include <string.h>

char* longestCommonPrefix(char** strs, int strsSize) {
    if (strsSize == 0) return "";

    char* pref = strs[0];
    int prefLen = strlen(pref);

    for (int i = 1; i < strsSize; i++) {
        char* s = strs[i];
        while (prefLen > strlen(s) || strncmp(pref, s, prefLen) != 0) {
            prefLen--;
            if (prefLen == 0) {
                return "";
            }
            pref[prefLen] = '\0';
        }
    }

    return pref;
}

File: problem 12/solution.cs

public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if (strs == null || strs.Length == 0) return "";

        string pref = strs[0];
        int prefLen = pref.Length;

        for (int i = 1; i < strs.Length; i++) {
            string s = strs[i];
            while (prefLen > s.Length || pref != s.Substring(0, prefLen)) {
                prefLen--;
                if (prefLen == 0) {
                    return "";
                }
                pref = pref.Substring(0, prefLen);
            }
        }

        return pref;
    }
}
